lib/ui/src/modules/ui/libs/hierarchy.js

```diff
@@ -46,24 +46,47 @@ function fillHierarchy(namespaces, hierarchy, story) {
   fillHierarchy(namespaces.slice(1), childHierarchy, story);
 }
 
-export function createHierarchy(stories) {
-  const hierarchyRoot = {
+export function createHierarchyRoot(name = '') {
+  return {
     isNamespace: true,
     namespaces: [],
-    name: '',
+    name,
     map: new Map(),
   };
+}
+
+export function createHierarchies(stories) {
+  const rootMap = {};
 
   if (stories) {
     stories.forEach(story => {
-      const { namespaces } = story;
+      const { rootName = '', namespaces } = story;
       const name = namespaces[namespaces.length - 1];
+      const hierarchyRoot =
+        rootMap[rootName] || (rootMap[rootName] = createHierarchyRoot(rootName));
 
       fillHierarchy(namespaces, hierarchyRoot, { ...story, name });
     });
   }
 
-  return hierarchyRoot;
+  return Object.keys(rootMap).map(rootName => rootMap[rootName]);
```

Comment:
`Object.values(rootMap)` ?

Reply:
Polyfills take care of `Object.values`?

Code is changed a bit at those lines though: I had to make sure that the main hierarchy is always the first in the array, so I didn't have use for `Object.values` anymore.

Which triggers the next question:
What could we do with the sorting of the other hierarchies in the panel? Now it respects the order in which the stories are loaded.